kolibri/plugins/learn/assets/src/actions.js

```diff
@@ -36,7 +36,35 @@ const fetchNodes = ({ dispatch }, id) => {
   });
 };
 
+
+const searchNodes = ({ dispatch }, params, page) => {
+  // Get the collection from ContentNodeResource.
+  const contentCollection = Kolibri.resources.ContentNodeResource.getCollection();
+  const wantedFields = ['pk', 'title', 'kind', 'instance_id', 'content_id', 'description', 'files'];
```

Comment:
The latency is due to the querying of the files field, though - removing the description somewhat reduces the payload size, but the latency stays roughly the same.

![image](https://cloud.githubusercontent.com/assets/1680573/16672041/39ebd6ca-4458-11e6-8294-7e4dee300f5a.png)